infrastructure/ansible/roles/flyway/tasks/main.yml

```diff
@@ -1,3 +1,41 @@
+- name: Create user on DB for application use
+  become: true
+  become_user: postgres
+  postgresql_user:
+    name: "{{ flyway_db_user }}"
+    password: "{{ flyway_db_password }}"
+    login_password: "{{ flyway_db_password }}"
+    encrypted: yes
+    state: "present"
+
+- name: Ensure database is created
+  become: true
+  become_user: postgres
+  postgresql_db:
+    name: "{{ flyway_db_name }}"
+    owner: "{{ flyway_db_user }}"
+
+- name: Ensure user has access to the database
+  become: true
+  become_user: postgres
+  postgresql_user:
+    db: "{{ flyway_db_name }}"
+    name: "{{ flyway_db_user }}"
+    password: "{{ flyway_db_password }}"
+    login_password: "{{ flyway_db_password }}"
+    encrypted: yes
+    priv: "ALL"
+    role_attr_flags: NOSUPERUSER,NOCREATEDB
+    state: "present"
+
+# This is a hack because the previous tasks were supposed to set
+# the DB level password for our user, but it typically does not.
+# To achieve this, we'll run the native SQL commands to set the
+# user DB password.
```

Comment:
So this is a vagrant bug?

Reply:
To be sure, this is not new, we're copying code from the old postgres role to the flyway role.

The problem is in the `postgresql_user` task. Ansible commands map to python scripts, the problem is in the [postgresql_user python script](
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/database/postgresql/postgresql_user.py).
Essentially the problem is the 'postgresql_user' task does not actually set the users password.